### PR TITLE
Corrected offests transformation in model.Differ when multiple change items interfere with each other

### LINKED ISSUE #1309: Differ: changes' offsets are incorrectly recalculated when multiple changes are done in one element
Original issue: https://github.com/ckeditor/ckeditor5/issues/849

The issue is caused by an incorrect change transformation. It happens when there are three changes in an element and:
- the second change interfered with first change,
- the third change interfered both with first change and the second change.

In that scenario, the second change has not been correctly transformed by the second change.

### PR DESCRIPTION
**Pull request by scofalik**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Fix: Corrected offests transformation in `model.Differ` when multiple change items interfere with each other. Closes #1309. Closes https://github.com/ckeditor/ckeditor5/issues/849.

### CODE REVIEW

File: src/model/differ.js [hunk 1]
```diff
@@ -547,6 +547,8 @@ export default class Differ {
 	 * @param {Array.<Object>} changes An array containing all the changes done on that element.
 	 */
 	_handleChange( inc, changes ) {
+		inc.newHowMany = inc.howMany;
```
Comment (pjasiun) [inline src/model/differ.js]: Some comment what `newHowMany` is and why it is needed is needed.

**Comment by pjasiun** [comment]
Missing tests.

**Comment by Mgsy** [comment]
After switching to this branch, there is the error in the console on init:
```
article.js:24 TypeError: editor.conversion.elementToElement is not a function
    at BlockQuoteEngine.init (blockquoteengine.js:42)
    at <anonymous>
```

**Comment by scofalik** [comment]
@pjasiun I will add a test.
@Mgsy are you sure you have all branches on `master`?

**Comment by Mgsy** [comment]
>@Mgsy are you sure you have all branches on master?

Yes, all branches are on `master`. I've removed all packages and installed it once again and the problem still exists.

**Comment by scofalik** [comment]
First, I had to refactor all the tests, because they were incorrect after engine refactoring. They worked by a coincidence.

Other than that, I rebased this branch with `master` and set all other branches on current `master`s. Tests are all green. I also added the missing test for this case.

**Comment by Mgsy** [comment]
The crash is gone, but data is restored incorrectly:

bug_ckeditor5

It will be a part of this PR or should I report it in the [Undo umbrella](https://github.com/ckeditor/ckeditor5-undo/issues/65)?

**Comment by scofalik** [comment]
Yes, I already reported the bug in the undo ticket.

**Pull request merged**